packages/widget/src/hooks/use-make-wallets.tsx

```diff
@@ -151,9 +158,30 @@ export const useMakeWallets = () => {
           continue;
         }
 
-        const minimalWallet: MinimalWallet = {
+        const evmGetAddress: MinimalWallet['getAddress'] = async ({
+          signRequired,
+          context,
+        }) => {
+          if (connector.id === currentConnector?.id) {
+            if (isEvmConnected && evmAddress) {
+              if (signRequired) {
+                trackWallet.track('evm', connector.id, chainType);
+              }
+
+              return evmAddress;
+            }
+          } else {
+            await connectAsync({ connector, chainId: Number(chainID) });
+            trackWallet.track('evm', connector.id, chainType);
+
+            return evmAddress;
+          }
+        };
```

Comment:
```suggestion
        const evmGetAddress: MinimalWallet['getAddress'] = async () => {
          if (connector.id !== currentConnector?.id) {
            await connectAsync({ connector, chainId: Number(chainID) });
            trackWallet.track('evm', connector.id, chainType);
          } else if (isEvmConnected && evmAddress & signRequired) {
            trackWallet.track('evm', connector.id, chainType);
          }
          return evmAddress;
        };
```

Reply:
Here I added the suggested again, you should be able to click commit suggestion now that it's not outdated